pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -526,7 +527,32 @@ func ringBufferMemoryUsage(mapStats *model.EBPFMapStats, info *ebpf.MapInfo, k *
 	return nil
 }
 
-func hashMapNumberOfEntriesWithBatch(mp *ebpf.Map) (int64, error) {
+type entryCountBuffers struct {
+	keys   []byte
+	values []byte
+	cursor []byte
+}
+
+func (e *entryCountBuffers) ensureSizeAll(referenceMap *ebpf.Map) {
+	maxSize := referenceMap.MaxEntries()
+	keysSize := referenceMap.KeySize() * maxSize
+	valuesSize := referenceMap.ValueSize() * maxSize
+	if uint32(cap(e.keys)) < keysSize {
```

Comment:
Is `cap` what the cilium library is using, and not `len`?

Reply:
The cilium library uses `len`, although from what I understand it makes sense to use `cap` here as we're interested in the buffer capacity and not just the length. However, in this case I think both will yield the same results.

Reply:
Do we ever trim the slices to less than its capacity? If we do, then using `cap` could be misleading because `len` would report less (and thus cilium might say it is undersized).

Reply:
We don't, but you're right, I changed it just in case